definitions/npm/jest_v21.x.x/flow_v0.104.x-/jest_v21.x.x.js

```diff
@@ -361,7 +361,7 @@ type JestObjectType = {
   * Returns the actual module instead of a mock, bypassing all checks on
   * whether the module should receive a mock implementation or not.
   */
- requireActual(moduleName: string): any,
+ requireActual<T>(m: $Flow$ModuleRef<T> | string): T,
```

Comment:
Could you explain a bit more in details how is this working, and what it's improving.

Reply:
I don't think I could explain it better than the [Flow commit that introduced `$Flow$ModuleRef<>`](https://github.com/facebook/flow/commit/296ed9ce7896790798b9b2fcb3ba8af7c43f7dbe) :-)

----

> Summary:
> This diff introduces a configurable string prefix that will upgrade any string literal in any position into a `$Flow$ModuleRef<$Exports<'TheModule'>>`. This lets currently hard-coded APIs like `requireLazy` be typed in userland, instead of being special-cased in the typechecker.
> 
> An example usage is a function like `ifRequired('m#Foo', Foo ==> { /* callback */ });`. This function talks to the underlying `require` internals to actually get the module exports and this new API would let this function be typed like so:
> 
> ```
> declare function ifRequired<TModule, TYes>(
>   id: $Flow$ModuleRef<TModule>,
>   cb: (module: TModule) => TYes,
> );
> ```
> 
> Flow would then understand that the first argument to the callback is in fact the same as requiring `Foo` directly. This reduces boilerplate and `typeof` in some of the magic infrastructural APIs.

----

W.r.t. `requireActual`, I've added an example of how it improves things in the PR summary.

Reply:
🏓

Reply:
Looks like this has broken my build.
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ flow-typed/npm/jest_v23.x.x.js:827:23

Cannot resolve name $Flow$ModuleRef.

     824│    * Returns the actual module instead of a mock, bypassing all checks on
     825│    * whether the module should receive a mock implementation or not.
     826│    */
     827│   requireActual<T>(m: $Flow$ModuleRef<T> | string): T,
     828│   /**
     829│    * Returns a mock module instead of the actual module, bypassing all checks
     830│    * on whether the module should be required normally or not.
```

I realize my flow version is sufficiently behind as I'm on `0.79.1`, we are looking to upgrade it soon.

Reply:
@kayhadrin @gwing33 Any taker for a fix PR?

Reply:
So if I understand correctly, you'd like to revert part of this PR so that only Flow v0.86.0+ is affected?

Reply:
@kayhadrin Yes, would be nice, scoping using file name versions.

Reply:
@kayhadrin @pascalduez Hi, is there a PR for reverting the changes for just Flow v0.86.0+? I have a build that's failing on v0.72.0 because of that change and I am looking to track it.

Reply:
I started to look at creating a PR, then got stuck with the `./build_and_test_cli.sh` command, I may take another stab tomorrow. A quick workaround I added in was just to `declare var $Flow$ModuleRef: any;`. We are in the process of upgrading flow, so it's short term for us.

I also just wanted to give a shout out to @pascalduez and @kayhadrin, thank you for your work and for what you're contributing to the community!